relayer/core/relayV2.go

```diff
@@ -18,12 +18,12 @@ func (re *RelayV2) Run() error {
 		return err
 	}
 
-	ethrelayerFactory := func() (workers.Worker, error) {
+	ethrelayerFactory := func() (workers.Worker, *workers.WorkerConfig, error) {
```

Comment:
I'd prefer if we pass config as a parameter to the factory function, instead of capturing it from the environment. This allows us to move the definition of these factories to to individual worker modules later down the line (that was always my intent when suggesting the factory idea)

Reply:
I tried that route at first but didn't want the `WorkerPool` to do any config parsing, i.e. when it selects a config entry to pass to a worker factory, it's "parsing" the config. I want to make config parsing the exclusive responsibility of relay.go.

An alternative that avoids config parsing in `WorkerPool` while also allowing the config to be passed to the factory: change `type WorkerPool []WorkerFactory` to `type WorkerPool [](WorkerFactory, WorkerConfig)`. What do you think?

Something worth noting: my initial instinct was also that the factory should accept the config as argument, but I changed my mind, specifically for `WorkerConfig` (other configs might be fine). Usage of params like `RestartDelay` and `Enabled` happen outside the worker and it's probably best if the worker itself has no knowledge of these concepts.

Reply:
yeh theres maybe 2 different kinds of config, one for how the pool should control the specific worker, and another internal to the worker